### PR TITLE
chore: add timeout configuration to env variables

### PR DESCRIPTION
**Pull request by balzdur**
Small refactor to allow configuring timeout with env vars.

### CODE REVIEW

File: api/server.go [hunk 1]
```diff
@@ -12,29 +12,55 @@ import (
...
 	var host string
-	if len(localTest) > 0 && localTest[0] {
+	if o.localTest {
 		host = "localhost"
 	} else {
 		host = "0.0.0.0"
 	}
 
+	// Add 5 seconds to the server timeout to gracefully handle the timeout in our code
+	maxTimeout := max(conf.BatchTimeout, conf.DecisionTimeout, conf.DefaultTimeout) + 5*time.Second
```
Comment (balzdur) [inline api/server.go:57]: This is something new (based on our spec discussion) : IIUC the previous comment comming from 
```go
// Infra timeout is 60sec, so we set it to 55sec in order to gracefully handle the timeout in our code
	BATCH_INGESTION_TIMEOUT = 55 * time.Second
```
This makes sense to me

File: api/routes.go [hunk 2]
```diff
@@ -22,21 +21,8 @@ func timeoutMiddleware(duration time.Duration) gin.HandlerFunc {
...
-	// TODO: temporary workaround, what we need to do is to make the endpoint reliably respond quickly even
-	// without warmup
-	LIST_DECISION_TIMEOUT = 30 * time.Second
-
-	DEFAULT_TIMEOUT = 5 * time.Second
-)
-
-func addRoutes(r *gin.Engine, auth Authentication, tokenHandler TokenHandler, uc usecases.Usecases, marbleAppHost string) {
-	tom := timeoutMiddleware(DEFAULT_TIMEOUT)
+func addRoutes(r *gin.Engine, conf Configuration, uc usecases.Usecases, auth Authentication, tokenHandler TokenHandler) {
```
[the remark this answers is not in the document]
Comment (Pascal-Delange) [reply api/routes.go:24]: no worries I'm ok, it's tightly coupled to the API anyway

File: api/routes.go [hunk 3]
```diff
@@ -48,19 +34,21 @@ func addRoutes(r *gin.Engine, auth Authentication, tokenHandler TokenHandler, uc
...
+	router.GET("/decisions", tom, handleListDecisions(uc, conf.MarbleAppHost))
+	router.GET("/decisions/with-ranks", tom,
+		handleListDecisionsInternal(uc, conf.MarbleAppHost))
+	router.POST("/decisions", timeoutMiddleware(conf.DecisionTimeout),
+		handlePostDecision(uc, conf.MarbleAppHost))
 	router.POST("/decisions/all",
-		timeoutMiddleware(SEQUENTIAL_DECISION_TIMEOUT),
-		handlePostAllDecisions(uc, marbleAppHost))
-	router.GET("/decisions/:decision_id", tom, handleGetDecision(uc, marbleAppHost))
+		timeoutMiddleware(3*conf.DefaultTimeout),
```
Comment (Pascal-Delange) [inline api/routes.go:43]:
```suggestion
		timeoutMiddleware(3*conf.DecisionTimeout),
```

File: api/routes.go [hunk 4]
```diff
@@ -48,19 +34,21 @@ func addRoutes(r *gin.Engine, auth Authentication, tokenHandler TokenHandler, uc
 
 	router.GET("/ast-expression/available-functions", tom, handleAvailableFunctions)
 
-	router.GET("/decisions", timeoutMiddleware(LIST_DECISION_TIMEOUT), handleListDecisions(uc, marbleAppHost))
-	router.GET("/decisions/with-ranks", tom, handleListDecisionsInternal(uc, marbleAppHost))
-	router.POST("/decisions", timeoutMiddleware(models.DECISION_TIMEOUT), handlePostDecision(uc, marbleAppHost))
+	router.GET("/decisions", tom, handleListDecisions(uc, conf.MarbleAppHost))
```
Comment (Pascal-Delange) [inline api/routes.go:37]: with decision timeout here

File: api/routes.go [hunk 5]
```diff
@@ -48,19 +34,21 @@ func addRoutes(r *gin.Engine, auth Authentication, tokenHandler TokenHandler, uc
...
+		timeoutMiddleware(3*conf.DefaultTimeout),
+		handlePostAllDecisions(uc, conf.MarbleAppHost))
+	router.GET("/decisions/:decision_id", tom, handleGetDecision(uc, conf.MarbleAppHost))
 	router.GET("/decisions/:decision_id/active-snoozes", tom, handleSnoozesOfDecision(uc))
 	router.POST("/decisions/:decision_id/snooze", tom, handleSnoozeDecision(uc))
 
 	router.POST("/ingestion/:object_type", tom, handleIngestion(uc))
 	router.POST("/ingestion/:object_type/multiple", tom, handleIngestionMultiple(uc))
-	router.POST("/ingestion/:object_type/batch", timeoutMiddleware(BATCH_INGESTION_TIMEOUT), handlePostCsvIngestion(uc))
+	router.POST("/ingestion/:object_type/batch", timeoutMiddleware(conf.BatchTimeout), handlePostCsvIngestion(uc))
```
[the remark this answers is not in the document]
Comment (Pascal-Delange) [reply api/routes.go:51]: oopsies, it was supposed to be the case in the decision_data_by_score/data_by_rule_execution endpoints. Let's add it there too !

File: api/server.go (hunk 1 above)
Comment (Pascal-Delange) [reply api/server.go:57]: Hum yea ok that works for me
The two kinds of timeout work on a different level but this should work fine

**Review by Pascal-Delange — APPROVED** [review]
Thanks ! A few minor suggestions and LGTM

**Pull request merged**